### PR TITLE
Fix 56, stop animation when limits are reached

### PR DESCRIPTION
**Pull request by mixcas**
Animation is stopped when limits are reached but triggers ALWAYS a new pan move is started.

Theres an alternative behaviour, where animation is not triggered when a new pan move is stared if new move is still reaching a limit. I left commented code for that and a note to remove that code later, depending on the desired behaviour.

Also makes some improvements like

- `Site.Map.window` moved to `Site.window`,  code adapted to use this new var name.
-  `Site.Map.getWindowSize()` moved to `Site.getWindowSize()`, bound to `Site.onResize` now

### CODE REVIEW

File: src/js/main.js [hunk 1]
```diff
@@ -2,12 +2,18 @@
...
 Site = {
-  $window: $(window),
   mobileThreshold: 1024,
   scrollSpeed: 300,
   init: function() {
     var _this = this;
 
+    _this.mapElement = document.getElementById('map');
+
+    _this.$window = $(window);
```
Comment (varffs) [inline src/js/main.js:12]: what difference is there between declaring on the object or at init?

File: src/js/main.js [hunk 2]
```diff
@@ -227,8 +269,8 @@ Site.Shapes = {
   playAnimation: function() {
     var _this = this;
 
-    // Check is animating has been stopped; we do this here to fail faster
-    if (!_this.animating) {
+    // Check is animating has been stopped or is at limit; we do this here to fail faster
+    if (!_this.animating || _this.atLimit) {
```
Comment (varffs) [inline src/js/main.js]: this line is not running when moving the map along one boundry. i.e. if you move all the way left then move up and down the left boundry the animation does not play and this function is never executed

**Review by varffs — COMMENTED** [review]
I'm afraid I don't understand the actions of the commented code and the branch doesn't seem to do what I think it should be doing (I could be wrong about that should be happening. Should the animation run when moving along a boundry?)

I think it would be better to refactor in refactor only branches as the window movement is muddling the PR changes, but that isn't the only reason I don't quite get it rn.

**Comment by mixcas** [comment]
@mikeyrayvon can u be super clear what are we doing here then? cuz i thought the animation was supposed to stop when we reached the edges

File: src/js/main.js (hunk 1 above)
Comment (mixcas) [reply src/js/main.js:12]: i only moved this for readability

File: src/js/main.js (hunk 2 above)
Comment (mixcas) [reply src/js/main.js]: yep, thats what i wanted it to do

**Comment by mixcas** [comment]
@varffs yeah, i know about the refactor changes. sorry about that, i had to fix that now cuz it was getting on the way. i always advocate for doing those on its own branch

File: src/js/main.js [hunk 3]
```diff
@@ -39,7 +65,7 @@ Site = {
   setIsMobileWidth: function() {
     var _this = this;
 
-    if (_this.$window.width() >= _this.mobileThreshold) {
+    if (_this.window.width >= _this.mobileThreshold) {
```
Comment (mikeyrayvon) [inline src/js/main.js:68]: im confused now. cause line 12 u declare `_this.$window` and then here and on line 48 & 49 u use`_this.window`

Comment (mixcas) [reply src/js/main.js:68]: yeah its kinda messy. theres `_this.$window` and `_this.window`. `_this.window` stores values such as window size and window center. this specific change is only a micro omprimization

File: src/js/main.js (hunk 2 above)
Comment (mikeyrayvon) [inline src/js/main.js]: though maybe we should be checking instead only for the corners and the middle of top, right, bottom, and left. where you actually are not panning, but in the pan zone. cause here youre looking for anywhere at the limit. I would say something more like `y == 0 && x <= Site.window.width * -2` for the top-right corner and then `x <= Site.window.width * -2 && y == Site.window.height / 2` for the middle of the right limit, etc? does that make sense? if would end up being 8 conditions (4 for the corners and 4 for the middle of each side), but maybe theres a way to simplify it...

Comment (mikeyrayvon) [inline src/js/main.js]: i should have been more explicit in my issue/card. sorry

File: src/js/main.js [hunk 4]
```diff
@@ -177,21 +203,37 @@ Site.Shapes = {
...
+    var x = event.detail.map.position.x;
+    var y = event.detail.map.position.y;
+
+    // Check for left limit
+    if (x >= 0 ||
+      x <= Site.window.width * -2 || // Check for right limit
+      y >= 0 || // Check for upper limit
+      y <= Site.window.height * -2) { // Check for bottom limit
+      _this.atLimit = true;
+    } else {
```
Comment (mikeyrayvon) [inline src/js/main.js]: though maybe we should be checking instead only for the corners and the middle of top, right, bottom, and left. where you actually are not panning, but in the pan zone. cause here youre looking for anywhere at the limit. I would say something more like `y == 0 && x <= Site.window.width * -2` for the top-right corner and then `x <= Site.window.width * -2 && y == Site.window.height / 2` for the middle of the right limit, etc? does that make sense? if would end up being 8 conditions (4 for the corners and 4 for the middle of each side), but maybe theres a way to simplify it...
i should have been more explicit in my issue/card. sorry

Comment (mixcas) [reply src/js/main.js]: both things are easy to do but centers would be hard to hit so they are kinda useless

**Comment by mixcas** [comment]
i think the best would be to think it better, what do we really want in terms of ux. [im thinking about it right now]

**Comment by mikeyrayvon** [comment]
they've put a lot of emphasis on having the background change feel like its tied to the map panning. like they're interdependent. so i think the idea is that when the map isn't panning, the background isnt changing, because one doesnt happen without the other.

**Comment by mikeyrayvon** [comment]
youre right its hard to hit the middles. so lets just look for the corners

**Comment by mixcas** [comment]
I think only the corners give a good feeling

**Pull request merged**